api/src/services/occurrence-service.ts

```diff
@@ -34,6 +57,140 @@ export class OccurrenceService extends DBService {
     this.occurrenceRepository = new OccurrenceRepository(connection);
   }
 
+  /**
+   * Get all occurrences within map bounds
+   *
+   * @param {(string | undefined)} [mapView]
+   * @return {*}  {Promise<IGetMapOccurrenceData[]>}
+   * @memberof OccurrenceService
+   */
+  async getMapOccurrences(mapView?: string | undefined): Promise<IGetMapOccurrenceData[]> {
+    const allOccurrences = await this.occurrenceRepository.getMapOccurrences(mapView);
+
+    const formated = this.formatOccurrenceDataForMap(allOccurrences);
+
+    return formated;
+  }
+
+  /**
+   * Format and curate raw data from occurrences
+   * First sort by same taxonid and geometry
+   *
+   * @param {IGetOccurrenceData[]} occurrenceData
+   * @return {*}  {IGetMapOccurrenceData[]}
+   * @memberof OccurrenceService
+   */
+  formatOccurrenceDataForMap(occurrenceData: IGetOccurrenceData[]): IGetMapOccurrenceData[] {
```

Comment:
For all of these functions, maybe add a few comments to each step explaining what you're doing and why. Ideally someone should be able to come along and refactor or modify this, and anyway we can help them understand the original intention the better.